src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Runtime/CompilerServices/RuntimeHelpersTests.cs

```diff
@@ -68,6 +68,15 @@ public static unsafe void GetObjectValue()
             Assert.Equal(i, (int)iOV);
         }
 
+        [Fact]
+        public static void EqualsTest()
+        {
+            // Boolean RuntimeHelpers.Equals(Object, Object)
+
+            Assert.True(RuntimeHelpers.Equals(Guid.Empty, Guid.Empty));
+            Assert.False(RuntimeHelpers.Equals(Guid.Empty, Guid.NewGuid()));
```

Comment:
```suggestion
            Guid guid = Guid.NewGuid();
            Assert.Equals(guid == Guid.Empty, RuntimeHelpers.Equals(Guid.Empty, guid));
```
In case we'd generate an empty GUID.

Reply:
`Guid.NewGuid` won't return empty.
The real thing I'm concerned is that we are taking a dependency of the behavior/contract of `NewGuid`.